### PR TITLE
feat: (levm) Log<N> operations

### LINKED ISSUE #501: Opcode: LOG<N>

### PR DESCRIPTION
**Pull request by belfortep**
**Motivation**

Adds the `log<n>` opcodes for the VM

**Description**

Adds the implementation of all the `log<n>`operations with tests for different cases
- create all the types of `logs`
- multiple `logs`
- logs in static calls

Closes #501

### CODE REVIEW

**Review by iinaki — APPROVED** [review]
good job :))

File: crates/levm/src/call_frame.rs [hunk 1]
```diff
@@ -2,6 +2,13 @@ use crate::{memory::Memory, opcodes::Opcode};
 use bytes::Bytes;
 use ethereum_types::{Address, U256};
 
+#[derive(Clone, Debug, Default, Eq, PartialEq, Hash)]
+pub struct Log {
+    pub address: Address,
+    pub topics: Vec<U256>,
```
Comment (lazcanoluca) [inline crates/levm/src/call_frame.rs:8]:
```suggestion
    pub topics: Vec<H32>,
```

I think the "correct" type to use here is H32: https://github.com/ethereum/execution-specs/blob/51fac24740e662844446439ceeb96a460aae0ba0/src/ethereum/cancun/blocks.py#L91

File: crates/levm/src/call_frame.rs [hunk 2]
```diff
@@ -2,6 +2,13 @@ use crate::{memory::Memory, opcodes::Opcode};
 use bytes::Bytes;
 use ethereum_types::{Address, U256};
 
+#[derive(Clone, Debug, Default, Eq, PartialEq, Hash)]
+pub struct Log {
```
Comment (lazcanoluca) [inline crates/levm/src/call_frame.rs:6]:
```suggestion
/// Data record produced during the execution of a transaction.
pub struct Log {
```

File: crates/levm/src/call_frame.rs (hunk 1 above)
Comment (belfortep) [reply crates/levm/src/call_frame.rs:8]: Oh you are right, just used the `U256` like the default, changing for that 🫡

**Review by lazcanoluca — APPROVED** [review]

File: crates/levm/src/call_frame.rs [hunk 3]
```diff
@@ -1,6 +1,14 @@
...
 use bytes::Bytes;
-use ethereum_types::{Address, U256};
+use ethereum_types::{Address, H32, U256};
+
+#[derive(Clone, Debug, Default, Eq, PartialEq, Hash)]
+/// Data record produced during the execution of a transaction.
+pub struct Log {
+    pub address: Address,
+    pub topics: Vec<H32>,
+    pub data: Vec<u8>,
```
Comment (ilitteri) [inline crates/levm/src/call_frame.rs:10]: Let's use `Bytes` to keep consistency.

```suggestion
    pub data: Bytes,
```

**Review by ilitteri — COMMENTED** [review]
LGTM. Left one comment.

**Review by ilitteri — APPROVED** [review]
LGTM

**Pull request merged**